csharp/ql/src/semmle/code/csharp/Chaining.qll

```diff
@@ -47,25 +41,24 @@ private predicate nonChainingBody(Callable c) {
 }
 
 private Expr returnedValue(Callable c) {
-  exists(Expr e |
-    c.canReturn(e) |
-    if e instanceof ConditionalExpr then
+  exists(Expr e | c.canReturn(e) |
+    if e instanceof ConditionalExpr
+    then
       result = e.(ConditionalExpr).getThen() or
       result = e.(ConditionalExpr).getElse()
-    else
-      result = e
+    else result = e
```

Comment:
@aschackmull Personally I think the `then` and `else` parts would look better balanced, i.e.
```
then
  result = e.(ConditionalExpr).getThen() or
  result = e.(ConditionalExpr).getElse()
else
  result = e
```

Reply:
This actually hits a fundamental limitation in the layout library that we use. Formatting if-then-else already covers a lot of different possibilities and it is not possible to make the space-or-newline after `then` correlated with the space-or-newline after `else` because of the space-or-newline before the `else`, which needs to sit in a correlation group further out.